src/models/training/t5_training_params.py

```diff
@@ -0,0 +1,141 @@
+        )
+        eval_set = ScienceQADatasetImg(
+            problems,
+            val_qids,
+            name_maps,
+            tokenizer,
+            args.input_len,
+            args.output_len,
+            args,
+            image_features,
+            args.eval_le,
+        )
+        test_set = ScienceQADatasetImg(
+            problems,
+            test_qids,
+            name_maps,
+            tokenizer,
+            args.input_len,
+            args.output_len,
+            args,
+            image_features,
+            args.test_le,
+        )
+    else:
+        train_set = ScienceQADatasetStd(
+            problems,
+            train_qids,
+            tokenizer,
+            args.input_len,
+            args.output_len,
+            args,
+        )
+        eval_set = ScienceQADatasetStd(
+            problems,
+            val_qids,
+            tokenizer,
+            args.input_len,
+            args.output_len,
+            args,
+            args.eval_le,
+        )
+
+        test_set = ScienceQADatasetStd(
+            problems,
+            test_qids,
+            tokenizer,
+            args.input_len,
+            args.output_len,
+            args,
+            args.test_le,
+        )
+
+    return train_set, eval_set, test_set
+
+
+def get_training_args(args, save_dir):
+    # only use the last model for evaluation to save time
+    if args.final_eval:
+        training_args = Seq2SeqTrainingArguments(
+            save_dir,
+            do_train=True if args.evaluate_dir is None else False,
+            do_eval=False,
+            evaluation_strategy="no",
+            logging_strategy="steps",
+            save_strategy="epoch",
+            save_total_limit=2,
+            learning_rate=args.lr,
+            eval_accumulation_steps=args.eval_acc,
+            per_device_train_batch_size=args.bs,
+            per_device_eval_batch_size=args.eval_bs,
+            weight_decay=0.01,
+            num_train_epochs=args.epoch,
+            predict_with_generate=args.use_generate,
+            report_to="none",
+        )
+    # evaluate at each epoch
+    else:
+        training_args = Seq2SeqTrainingArguments(
+            save_dir,
+            do_train=True if args.evaluate_dir is None else False,
+            do_eval=True,
+            evaluation_strategy="epoch",
+            logging_strategy="steps",
+            save_strategy="epoch",
+            save_total_limit=2,
+            learning_rate=args.lr,
+            eval_accumulation_steps=args.eval_acc,
+            per_device_train_batch_size=args.bs,
+            per_device_eval_batch_size=args.eval_bs,
+            weight_decay=0.01,
+            num_train_epochs=args.epoch,
+            metric_for_best_model="accuracy" if args.prompt_format != "QCM-LE" else "rougeL",
+            predict_with_generate=args.use_generate,
+            load_best_model_at_end=True,
+            report_to="none",
+        )
+    return training_args
```

Comment:
This is ugly...
Let's make a builder which takes the common parameters and we set only the parameters that differ (if better solutions don't come up)